scRNA-seq/02-normalizing_scRNA-seq.Rmd

```diff
@@ -67,185 +66,298 @@ sc_metadata <- readr::read_tsv(metadata_file)
+sample_info <- tibble::tibble(geo_accession = colnames(gbm_sce)) %>%
+  #join to the metadata
+  dplyr::left_join(sc_metadata) %>%
+  dplyr::select(geo_accession,
+                title,
+                cell_type = cell.type.ch1,
+                tissue = characteristics_ch1.3
+                ) %>%
+  # clean up tissue
+  dplyr::mutate(tissue = stringr::str_remove(tissue, "tissue: "))
+```
+
+Check that the sample info accession ids are the same as the columns of our data
+
+```{r check_sampleinfo, live = TRUE}
+all.equal(sample_info$geo_accession, colnames(gbm_sce))
+```
+
+Now add that data to the SCE object.
+For unclear reasons, we have to convert our table to a `DataFrame` object in order for this to work.
+Just to keep things confusing, a `DataFrame` is not the same as a `data.frame` that we have been using throughout. 
+We also need to be sure to include the `row.names` argument to keep those properly attached
+
+```{r}
+colData(gbm_sce) <- DataFrame(sample_info, row.names = sample_info$geo_accession)
+```
+
 
 ## Normalization of count data 
 
-In whatever data we are working with, we are always looking to maximize biological 
-variance and minimize technical variance. 
+In whatever data we are working with, we are always looking to maximize biological variance and minimize technical variance. 
+A primary source of technical variation we are concerned with is the variation in library sizes among our samples. 
+While different cells may have different total transcript counts, it seems far more likely that the primary source of variation that we see is due to library construction, amplification, and sequencing. 
+
 This is where normalization methods usually come into the workflow.
-With this high amount of zeroes in our data, this makes normalization particularly
-tricky.
-So in order to account for these zeroes, we normalize cells in groups with other 
-cells like them; a method introduced in [Lun et al 2016](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0947-7).
+The distribution of the counts that we saw in the previous notebook, and in particular the fact that the count data is noisy with many zero counts, makes normalization particularly tricky.
+To handle this noise, we normalize cells in groups with other cells like them; a method introduced in [Lun *et al.* (2016)](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0947-7).
+
+Briefly, we first cluster the cells to find groups of similar cells, then compute normalization factors based on the sums of expression in those groups. 
+The group normalization is then applied back to the individual cells within the group to create a normalized count matrix. 
+In this case, we will also log-transform the normalized counts to get a less skewed distribution of expression measures. 
+Note that because of the zero counts, the logNormCounts() function will add a pseudocount of 1 to each value before computing the log.
```

Comment:
```suggestion
Note that because of the zero counts, the `logNormCounts()` function will add a pseudocount of 1 to each value before computing the log.
```